app/MMB/src/scenes/passenger/menu/TitledMenuGroup.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import * as React from 'react';
+import { Platform } from 'react-native';
+import { Translation } from '@kiwicom/mobile-localization';
+
+import MenuGroup from './MenuGroup';
+import MenuGroupTitle from '../../../components/menu/MenuGroupTitle';
+
+type Props = {|
+  title: React.Element<typeof Translation>,
+  children: React.Node,
+|};
+
+// This is basically a copy of /components/TitledMenuGroup
+// But it needs a different type of separator, so it uses ad different MenuGroup to wrap its children
```

Comment:
I don't like these copy-pastes. It's far from being optimal. 😞 I would actually prefer the code before over this. 😞 I think you are working on this, right?

![screen shot 2018-05-25 at 09 52 44](https://user-images.githubusercontent.com/978368/40532954-6b3a0212-6001-11e8-88ed-b406b4651a2a.png)

I expected something like this:

- RED: titled group is the same (it will have only one element and one responsibility - do the title and card styles)
- BLUE: one new child that will create these multibuttons (double, tripe with one arrow) - you don't have to deal with separators here because there is going to be only one child (in the RED)
- GREEN: new menu item with the new styles (you could decompose `MenuItem` and partially reuse it)

![screen shot 2018-05-25 at 09 52 44](https://user-images.githubusercontent.com/978368/40533168-195bede2-6002-11e8-91ae-d93a93df7476.png)

Do you want to help with the design here?

Reply:
Well, I am a little bit lost here, If I should not change the existing components, and not copy it's behaviour and not create new one, I don't really know what to do.

Reply:
> I should not change the existing components, and not copy it's behaviour and not create new one

Yes, this is what I just wrote. 🙄 It's fine to modify and reuse the components and I literally wrote I expected new components. I just don't think that the right way how to modify the components is to add new props for every internal element like `titleStyle`, `descStyle`... (see: https://github.com/kiwicom/mobile/pull/640)

Wanna do pair programming for this?

Reply:
> Wanna do pair programming for this?
sure 🙂